docs/beamrunner/beamrunner-3-sample.md

```diff
@@ -0,0 +1,172 @@
+                if (temp < badTempThreshold) {
+                    // Good reading, output and update distribution metrics
+                    c.output(c.element());
+                    ...
+                } else {
+                    // Bad reading, output to side output, update counters
+                    c.output(badTag, c.element());
+                    ...
+                }
+            }
+        }).withOutputTags(goodTag, TupleTagList.of(badTag)));
+
+
+```
+
+On the good readings, statistics are calculated in a GoodStats transform and then logged by a GoodLog transform:
+
+```
+validatedReadings
+        .get(goodTag)
+        .apply("GoodStats", Combine.<String, Double, Stats>perKey(new DeviceStatsFn()))
+        .apply("GoodLog", ParDo.of(new LogWindowFn<Stats>("Temperature device statistics")));
+```
+
+The bad readings are counted by a BadCount transform and logged by the BadLog transform:
+
+```
+validatedReadings
+        .get(badTag)
+        .apply("BadCount", Count.<String,Double>perKey())
+        .apply("BadLog", ParDo.of(new LogWindowFn<Long>("Temperature device error count")));
+```
+
+The structure of the application becomes clear after it is displayed in the Streams console after the application is run.
+
+## Running the `TemperatureSample` application
+
+1. Make sure that your Streaming Analytics service on Bluemix is running.
+2. Navigate to the `$STREAMS\_RUNNER\_HOME/samples` directory. The Streams Runner toolkit provides all necessary files. Assuming that all environment variables are set as described in Installing the toolkit and that $VCAP\_SERVICES has credentials in it named beam-service, you can launch the `TemperatureSample` application with the following command:
+
+   ```
+   java -cp \
+   $STREAMS_BEAM_TOOLKIT/lib/com.ibm.streams.beam.translation.jar:\
+   $STREAMS_RUNNER_HOME/samples/lib/com.ibm.streams.beam.samples.jar \
+   com.ibm.streams.beam.sample.temperature.TemperatureSample \
+   --runner=StreamsRunner \
+   --contextType=STREAMING_ANALYTICS_SERVICE \
+   --jarsToStage=$STREAMS_RUNNER_HOME/samples/lib/com.ibm.streams.beam.samples.jar```
+
+   Note: If the environment variables are not set, you must use full paths, and use the `--vcapServices` parameter to provide the path to the Bluemix credentials file.
+
+   For more information about the parameters available to you, see [Pipeline options for Streams Runner](../beamrunner-5-ref/#pipeline-options-for-streams-runner).
+
+3. Verify that the application started successfully.
+
+   Streams Runner displays information (INFO) messages in the terminal as it processes the Beam pipeline for your application, turning it into an IBM Streams application bundle (.sab) file, and eventually submitting the .sab file to IBM Bluemix:
+
+   ```
+   Sep 27, 2017 2:34:19 PM com.ibm.streams.beam.translation.StreamsRunner run
+   INFO: Running pipeline using StreamsRunner in context:
+        contextType=STREAMING_ANALYTICS_SERVICE

```

Comment:
Minor thing: how come the values for `beamToolkitDir` is null? That seems unusual since we set the environment variable (just from my novice perspective)

Reply:
@natashadsilva Good point. We print out this info before we do any processing (looking up the toolkit location). I'll add an issue to migrate this somewhere else. (This gets a bit detailed but this is under INFO logging level which is not the default logging level so this info could disappear).